bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -52,73 +64,78 @@ def optimizedparam(sampleval, nfold=2):
 
-#calling signal data and constructing signal_kde_hdf file over template params
+#Obtaining signal parameters
+signal_pars = []
 if args.make_signal_kde:
-    data_signal = numpy.genfromtxt(args.signal_file, dtype = float, delimiter=',', names = True)
-    mass2 = data_signal['mass2']
-    N_original = len(mass2)
+    signal_file = numpy.genfromtxt(args.signal_file, dtype = float, delimiter=',', names = True)
+    mass2_sgnl = signal_file['mass2']
+    N_original = len(mass2_sgnl)
     if args.min_mass:
-        idx = mass2 > args.min_mass
-        mass2 = mass2[idx]
+        idx = mass2_sgnl > args.min_mass
+        mass2_sgnl = mass2_sgnl[idx]
         logging.info('%i triggers out of %i with MASS2 > %s' %
-                         (len(mass2), N_original, str(args.min_mass)))
+                         (len(mass2_sgnl), N_original, str(args.min_mass)))
     else:
         idx = numpy.full(N_original, True)
-    mass2_s = mass2
-    mass1_s = data_signal['mass1'][idx]
-    assert min(mass1_s - mass2_s) > 0
-    eta_s = data_signal['eta'][idx]
-    chi_eff_s  = data_signal['chi_eff'][idx]
-    mchirp_s = data_signal['mchirp'][idx]
+    mass1_sgnl = signal_file['mass1'][idx]
+    assert min(mass1_sgnl - mass2_sgnl) > 0
+
+    for param, slog in zip(args.fit_param, args.log_param):
+        data = signal_file[param][idx]
+        if slog in ['False']:
+            logging.info('Using param: %s', param)
+            signal_pars.append(data)
+        elif slog in ['True']:
+            logging.info('Using log param: %s', param)
+            signal_pars.append(numpy.log(data))
+        else:
+            raise ValueError("invalid log param argument, use 'True', or 'False'")
 
-    sample_s = numpy.vstack((numpy.log(mchirp_s), eta_s, chi_eff_s)).T
-    grid_pts_s = numpy.array((numpy.log(mchirp_n), eta_n, chi_eff_n)).T
+    signal_samples = numpy.vstack((signal_pars)).T
     logging.info('Starting optimization of signal KDE parameters')
-    optbw_s, optalpha_s = optimizedparam(sample_s, nfold=2)
+    optbw, optalpha = optimizedparam(signal_samples, nfold=eval(args.nfold_signal))
     logging.info('Evaluating signal KDE')
-    kde_data_s = kde_awkde(sample_s, grid_pts_s, alp=optalpha_s, gl_bandwidth=optbw_s)
- 
-    hdf = h5py.File(args.output_file, 'w')
-    hdf.create_dataset('kde_sig', data=kde_data_s)
-    hdf.attrs['signal_kde'] = 'kde_sig'
-    hdf.create_dataset('mass1', data=mass1_s)
-    hdf.create_dataset('mass2', data=mass2_s)
-    hdf.create_dataset('mchirp', data=mchirp_s)
-    hdf.create_dataset('eta', data=eta_s)
-    hdf.create_dataset('chi_eff', data=chi_eff_s)
-    hdf.create_dataset('grid_pts', data=grid_pts_s)
-    hdf.close()
+    signal_kde = kde_awkde(signal_samples, temp_samples, alp=optalpha, gl_bandwidth=optbw)
+
+    with h5py.File(args.output_file, "w") as f_dest:
+        f_dest.create_dataset("signal_kde", data=signal_kde)
+        for name in args.fit_param:
```

Comment:
Is it possible to not write these pieces of code for outputting the fit param values and original template file values to the output hdf file twice?  Since you are doing the same for the template kde and signal kde cases, can this be done with only one piece of code?